code/datums/components/mood.dm

```diff
@@ -147,17 +158,29 @@
 			add_event("jolly", /datum/mood_event/jolly)
 			clear_event("depression")
 
-/datum/component/mood/proc/DecreaseSanity(amount, limit = 0)
-	if(sanity < limit) //This might make KevinZ stop fucking pinging me.
+	holdmyinsanityeffect = insanity_effect
+
+/datum/component/mood/proc/DecreaseSanity(amount, minimum = SANITY_INSANE)
+	if(sanity < minimum) //This might make KevinZ stop fucking pinging me.
 		IncreaseSanity(0.5)
 	else
-		sanity = max(0, sanity - amount)
-
-/datum/component/mood/proc/IncreaseSanity(amount, limit = 99)
-	if(sanity > limit)
+		sanity = max(minimum, sanity - amount)
+		if(sanity < SANITY_UNSTABLE)
+			if(sanity < SANITY_CRAZY)
+				insanity_effect = (MAJOR_INSANITY_PEN)
+			else
+				insanity_effect = (MINOR_INSANITY_PEN)
+
+/datum/component/mood/proc/IncreaseSanity(amount, maximum = SANITY_NEUTRAL)
+	if(sanity > maximum)
 		DecreaseSanity(0.5) //Removes some sanity to go back to our current limit.
```

Comment:
I really have no idea how this system works but decreasing (or increasing in the other proc by arbitrary amount) seems like very bad idea if you know what value you want.

Reply:
the maximum sanity (or minimum) is a dynamic level based on mood, so we don't know what value we want w/o checking for mood level (which is what governs this proc's arguments in process).

Reply:
Or the answer is "it's to simulate a gradual incline/decline to the point versus just setting it instantly since mood shifts"